src/js/rtc_session.js

```diff
@@ -917,12 +917,32 @@ export default class RtcSession {
             }
 
             return await Promise.all(tracks.map(async (track) => {
-                var rawStats = await this._pc.getStats(track);
-                var digestedStats = extractMediaStatsFromStats(timestamp, rawStats, streamType);
-                if (! digestedStats) {
-                    throw new Error('Failed to extract MediaRtpStats from RTCStatsReport for stream type ' + streamType);
+                // get legacy stats report as a promise
+                if (navigator.webkitGetUserMedia) {
```

Comment:
any better way to determine if the legacy interface is still available?

Reply:
Added a function to determine if legacy interface is supported.